addons/rose/addon/components/rose/loading-button/index.js

```diff
@@ -0,0 +1,20 @@
+import button from '../button/index';
+
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class LoadingButton extends button {
+  // =actions
+  @tracked isLoading = false;
+
+  @action
+  async toggleRefresh() {
+    this.isLoading = true;
+    try {
+      await this.args.onClick();
+    } catch (e) {
+      console.error('Error while loading data', e);
```

Comment:
Do we generally do any sort of error handling for this? e.g. Displaying an alert that something failed?